pyoptsparse/pySNOPT/pySNOPT.py

```diff
@@ -436,22 +473,23 @@ def __call__(
 
             Names = np.array(["        "], "c")
             pi = np.zeros(ncon, np.float)
-            rc = np.zeros(nvar + ncon, np.float)
             inform = np.array([-1], np.intc)
             mincw = np.array([0], np.intc)
             miniw = np.array([0], np.intc)
             minrw = np.array([0], np.intc)
-            nS = np.array([0], np.intc)
-            ninf = np.array([0], np.intc)
-            sinf = np.array([0.0], np.float)
 
+            if restartDict is not None:
+                hs = restartDict["hs"]
```

Comment:
We could add an inline comment, just to specify what these SNOPT parameters are

Reply:
Sure, these constitute the "current state", so we pass them into SNOPT when calling `snkerc`. This is also why they are set differently from the work arrays.